packages/backend-tasks/src/tasks/TaskWorker.ts

```diff
@@ -123,23 +126,46 @@ export class TaskWorker {
   /**
    * Perform the initial store of the task info
    */
-  async persistTask(settings: TaskSettingsV1) {
+  async persistTask(settings: TaskSettingsV2) {
     // Perform an initial parse to ensure that we will definitely be able to
     // read it back again.
-    taskSettingsV1Schema.parse(settings);
+    taskSettingsV2Schema.parse(settings);
+
+    let startAt: DateTime = settings.initialDelayDuration
+      ? DateTime.now().plus(Duration.fromISO(settings.initialDelayDuration))
+      : DateTime.now();
+
+    const isCron = !settings?.cadence.startsWith('P');
+
+    // if we don't have a delay setting, and this is a crontab configuration, then the next start time is simply the
+    // time cron should trigger he job.
+    if (!settings.initialDelayDuration && isCron) {
+      const n = settings as TaskSettingsV2;
+      const time = new CronTime(n.cadence).sendAt().toISOString();
+      // start at the next cadence, sub one second to call the function immediately if we're using "* * * * * *"
+      startAt = DateTime.fromISO(time).minus(
+        Duration.fromObject({ seconds: 1 }),
+      );
```

Comment:
I'm torn about all this, for only one reason - it used to be that the database was the clock source. By which I mean, the code was carefully crafted around making the db figure out what "now" means etc, such that clock drift among several collaborating hosts never could become a problem. As it currently stands, the code now relies on the host to compute timestamps.

I realize that this probably will be very hard to sustain at this point. We could maintain the old code for the non-cron case I guess, but ... maybe it's not worth it.

Reply:
At least for postgres, you would typically have a single write master at least. It can be argued that these kinds of operations should always be made against a special pool that only ever connects to the master even for read ops. But it may not be necessary - these are stored as fixed timestamps that are read verbatim and transferred to the client without being interpreted by the reading db host as far as I know